orttraining/orttraining/training_ops/cuda/nn/dropout.cc

```diff
@@ -47,11 +48,30 @@ Status Dropout<T1, T2>::ComputeInternal(OpKernelContext* context) const {
 
   //Get Y_data
   auto Y = context->Output(0, shape);
+  ORT_ENFORCE(Y->Shape() == shape, "X and Y should have the same shape");
```

Comment:
The check seems redundant... 
Y is created according to shape one line above ... 

Usually, we check the shape for inputs if there is any requirement